snapshot-jms-test/src/main/java/com/hazelcast/jet/tests/snapshot/jms/JmsMessageProducer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.snapshot.jms;
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+
+import javax.jms.Connection;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+class JmsMessageProducer {
```

Comment:
this class looks like the same as the `jms-test` producer. I think we can depend/shade it into the test jar like we do with the `soak-tests-common`

Reply:
You're right, they are almost the same. I will create an issue for that and fix it after release. Is it ok?

Reply:
👍